src/metrics/peerstore.go

```diff
@@ -359,6 +361,41 @@ func (c *PeerStore) ExportToCSV(filePath string) error {
 	return nil
 }
 
+// Function that save the raw Data of all the peers to a db (Using BoltDB)
+// Using the peerID as a key, and the Peer struct marshalled into JSON as a value
+func (c *PeerStore) PersistToDb() {
+	db, err := bolt.Open("../../peerstore.db", 0600, &bolt.Options{Timeout: 5 * time.Second})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	err = db.Update(func(tx *bolt.Tx) error {
+		b, err := tx.CreateBucketIfNotExists([]byte("PeerStore"))
```

Comment:
What happens if you:
* run it multiple times with different peers in the peerstore
* run it with x stored peers and then no peers
* change the information of a given peer

just trying to understand how it works under the hood. does it append it to the existing database? does it update the value of a given key?

Reply:
- If peers have the peer.ID, it would overwrite their data. The idea was that in the future each peerID as a key have a bucket as a Value. And then this bucket would have timestamp as key and the JSON data as value so we never overwrite past data.
- My guess is nothing would happen if there are no peers, but will make sure by adding this case to the unit test
- for now it would overwrite it
Yes it would update the value of a given key

Reply:
cool, resolving